internal/controller/compute/statefulserverset/statefulserverset.go

```diff
@@ -216,18 +216,20 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 
 func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
 	cr, ok := mg.(*v1alpha1.StatefulServerSet)
-	cr.SetConditions(xpv1.Deleting())
 	if !ok {
 		return errors.New(errNotStatefulServerSet)
 	}
 
-	e.log.Info("Deleting the Data Volumes with label", "label", cr.Name)
+	cr.SetConditions(xpv1.Deleting())
+	meta.SetExternalName(cr, "")
```

Comment:
Are we sure this is necessary? It is possible that removing the external name annotation here before we know `Delete` is actually successful might trigger a new unintended creation on reconcile requeue.

Reply:
I can remove it. I took as example the code from the `serverset.go` and though that it would have been needed here as well